pkg/controller/job/job_controller.go

```diff
@@ -494,8 +496,12 @@ func (jm *JobController) syncJob(key string) (bool, error) {
 	var failureReason string
 	var failureMessage string
 
-	jobHaveNewFailure := failed > job.Status.Failed
-	exceedsBackoffLimit := jobHaveNewFailure && (int32(previousRetry)+1 > *job.Spec.BackoffLimit)
+	jobHaveNewFailure := (failed > job.Status.Failed)
```

Comment:
This could happen when Job Status failed to be updated?

Reply:
This did not change, it is as it was before I was trying different scenarios, so that change is nothing new. During tests I've just added parenthesis here.